src/services/car/__tests__/messageLead.test.ts

```diff
@@ -364,6 +367,80 @@ describe("Car API", () => {
     })
   })
 
+  describe("#fetchDealerWhatsAppLeads", () => {
+    const { content, pagination } = PaginatedLeads([
+      SearchWhatsAppLeadFactory(),
+    ])
+
+    beforeEach(() => {
+      fetchMock.mockClear()
+      fetchMock.mockResponse(
+        JSON.stringify({
+          content: content,
+          ...pagination,
+        })
+      )
+    })
+
+    it("calls correct endpoint", async () => {
+      await fetchDealerWhatsAppLeads({
+        dealerId: 1234,
+        query: {
+          page: 2,
+          size: 7,
+          sort: defaultLeadSort,
+        },
+        options: {
+          accessToken: "DummyTokenString",
+        },
+      })
+
+      expect(fetch).toHaveBeenCalledWith(
+        expect.stringMatching(
+          /\/dealers\/1234\/whats-app-tracking-entries\?page=1&size=7&sort=createdDate%2Cdesc$/
+        ),
+        expect.any(Object)
+      )
+    })
+
+    it("should trow error if accessToken is not passed", async () => {
+      let error
+      try {
+        await fetchDealerWhatsAppLeads({
+          dealerId: 1234,
+          query: {
+            page: 0,
+            size: 7,
+            sort: {},
+          },
+        })
+      } catch (err) {
+        error = err
+      }
+
+      expect(fetch).not.toHaveBeenCalled()
+      expect(error).toBeDefined()
```

Comment:
I'm not 100% sure here but I think you can move the two error asserts into the catch block then you can remove the let variable at the top

Reply:
I tried but it doesn't work.

Reply:
checked out your branch. Works perfectly for me:
```javascript
it("should trow error if accessToken is not passed", async () => {
      try {
        await fetchDealerMessageLeads({
          dealerId: 1234,
          query: {
            page: 0,
            size: 7,
            sort: {},
            searchQuery: "",
          },
        })
        expect(fetch).not.toHaveBeenCalled()
      } catch (err) {
        expect(err).toBeDefined()
        expect(err.message).toBeDefined()
      }
    })
```